src/server/scripts/Zandalar/Underrot/boss_elder_leaxa.cpp

```diff
@@ -0,0 +1,385 @@
+                    DoCastAOE(SPELL_CREEPING_ROT_SELECTOR);
+                    events.ScheduleEvent(EVENT_CREEPING_ROT, 16s);
+                    break;
+                case EVENT_BLOOD_MIRROR:
+                    Talk(SAY_BLOOD_MIRROR);
+                    Talk(SAY_ANNOUNCE_BLOOD_MIRROR);
+                    DoCastSelf(SPELL_BLOOD_MIRROR_SELECTOR);
+                    events.ScheduleEvent(EVENT_BLOOD_MIRROR, 47s);
+                    break;
+                default:
+                    break;
+            }
+
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
+        }
+
+        DoMeleeAttackIfReady();
+    }
+};
+
+// 134701 - Blood Effigy
+struct npc_blood_effigy : public ScriptedAI
+{
+    npc_blood_effigy(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        _events.ScheduleEvent(EVENT_BLOOD_BOLT, 1s);
+        _events.ScheduleEvent(EVENT_SANGUINE_FEAST, 8s);
+        _events.ScheduleEvent(EVENT_CREEPING_ROT, 12s);
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        me->SetAIAnimKitId(BLOOD_EFFIGY_DEATH);
+        me->DespawnOrUnsummon(3s);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        if (!UpdateVictim())
+            return;
+
+        _events.Update(diff);
+
+        if (me->HasUnitState(UNIT_STATE_CASTING))
+            return;
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_BLOOD_BOLT:
+                    DoCastVictim(SPELL_BLOOD_BOLT);
+                    _events.ScheduleEvent(EVENT_BLOOD_BOLT, 2s);
+                    break;
+                case EVENT_SANGUINE_FEAST:
+                    DoCastAOE(SPELL_SANGUINE_FEAST);
+                    _events.ScheduleEvent(EVENT_SANGUINE_FEAST, 30s);
+                    break;
+                case EVENT_CREEPING_ROT:
+                    DoCastAOE(SPELL_CREEPING_ROT_SELECTOR);
+                    _events.ScheduleEvent(EVENT_CREEPING_ROT, 16s);
+                    break;
+                default:
+                    break;
+            }
+
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
+        }
+
+        DoMeleeAttackIfReady();
+    }
+private:
+    EventMap _events;
+};
+
+// 132398 - Blood Wave Stalker
+struct npc_blood_wave_stalker : public ScriptedAI
+{
+    npc_blood_wave_stalker(Creature* creature) : ScriptedAI(creature) { }
+
+    static constexpr float WALK_DISTANCE = 58.0f;
+
+    void JustAppeared() override
+    {
+        DoCast(SPELL_CREEPING_ROT_AURA);
+
+        float angle = me->GetOrientation();
+        float destX = me->GetPositionX() + WALK_DISTANCE * std::cos(angle);
+        float destY = me->GetPositionY() + WALK_DISTANCE * std::sin(angle);
+        float destZ = me->GetPositionZ();
+        Position walkDest(destX, destY, destZ);
+
+        me->GetMotionMaster()->MovePoint(0, walkDest);
```

Comment:
Magic number, call it whatever </3

Reply:
that point is not used for anything, 0 is fine